src/core/Dropdown/Dropdown/Dropdown.tsx

```diff
@@ -241,37 +177,60 @@ export class Dropdown extends Component<DropdownProps> {
         id={id}
         {...passProps}
       >
-        <HtmlLabel
-          id={labelId}
-          {...labelProps}
-          className={dropdownClassNames.label}
-        >
-          {labelMode === 'hidden' ? (
-            <VisuallyHidden>{labelText}</VisuallyHidden>
-          ) : (
-            <Paragraph {...labelTextProps}>{labelText}</Paragraph>
-          )}
-        </HtmlLabel>
-        <ListboxInput {...listboxInputProps}>
-          <ListboxButton {...passDropdownButtonProps}>
-            {listboxDisplayValue}
-          </ListboxButton>
-          {!!ListboxPopoverComponentReplace ? (
-            <ListboxPopoverComponentReplace {...passDropdownPopoverProps}>
-              {this.dropdownItems(children, passDropdownItemProps)}
-            </ListboxPopoverComponentReplace>
-          ) : (
+        <HtmlDiv>
+          <LabelText
+            id={labelId}
+            labelMode={labelMode}
+            as="label"
+            optionalText={optionalText}
+          >
+            {labelText}
+          </LabelText>
+          <ListboxInput {...listboxInputProps}>
+            <ListboxButton
+              {...dropdownButtonProps}
+              id={buttonId}
+              className={classnames(dropdownClassNames.button, {
+                [dropdownClassNames.disabled]: !!disabled,
+              })}
+              {...buttonAriaLabelledByOverride}
+            >
+              {listboxDisplayValue}
+            </ListboxButton>
             <ListboxPopover
               position={positionMatchWidth}
               {...passDropdownPopoverProps}
             >
-              <ListboxList>
-                {this.dropdownItems(children, passDropdownItemProps)}
-              </ListboxList>
+              <ListboxList>{children}</ListboxList>
             </ListboxPopover>
-          )}
-        </ListboxInput>
+          </ListboxInput>
+        </HtmlDiv>
       </HtmlSpan>
     );
   }
 }
+
+const StyledDropdown = styled(
+  ({
+    tokens,
+    id: propId,
+    ...passProps
+  }: DropdownProps & InternalTokensProp) => (
+    <AutoId id={propId}>
```

Comment:
Could we move this to top level Dropdown? This way styled would only handle styles and the provider would not have any effect on the components below => a bit cleaner solution overall.

Reply:
Good point. Moved it there.

Reply:
Actually there were some repercussions from that, and as I took a closer look, it is in the styled variant in all the other implementations as well. Moved it back there in this one as well.

Reply:
It is there for other implementations with the exception of RadioButton. It was moved there to have a cleaner implementations. There shoud be no obstacles for doing so here as well.

Reply:
Now on top level.